sklearn/model_selection/_search.py

```diff
@@ -575,7 +587,26 @@ def fit(self, X, y=None, groups=None, **fit_params):
                 fit_params = self.fit_params
         estimator = self.estimator
         cv = check_cv(self.cv, y, classifier=is_classifier(estimator))
-        self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
+
+        self.scorer_, self.multimetric_ = _check_multimetric_scoring(
+            self.estimator, scoring=self.scoring)
+
+        if self.multimetric_:
+            if self.refit is not False and (
+                    not isinstance(self.refit, six.string_types) or
+                    # This will work for both dict / list (tuple)
+                    self.refit not in self.scorer_):
+                raise ValueError("For multi-metric scoring, the parameter "
+                                 "refit must be set to a scorer key "
+                                 "to make the best_* attributes "
```

Comment:
to refit an estimator with the best parameter setting on the whole data and make ...